tests/test_project.py

```diff
@@ -806,6 +807,28 @@ def test_update_submodules_list(repos_tmpdir):
                                 capture_stderr=True, capture_stdout=True)
     assert not (res.returncode or res.stdout.strip())
 
+    # Test freeze output with submodules
+    # see test_manifest_freeze for details
+    actual = cmd('manifest --freeze', cwd=ws).splitlines()
+    expected_res = ['^manifest:$',
+                    '^  projects:$',
+                    '^  - name: zephyr$',
+                    f'^    url: {re.escape(str(zephyr))}$',
+                    '^    revision: [a-f0-9]{40}$',
+                    '^    submodules:$',
+                    '^    - path: tagged_repo$',
+                    '^  - name: net-tools$',
+                    f'^    url: {re.escape(str(net_tools))}$',
+                    '^    revision: [a-f0-9]{40}$',
+                    '^    submodules:$',
+                    f'^    - path: {re.escape(str(kconfiglib_submodule))}$',
+                    '^      name: Kconfiglib$',
+                    '^  self:$',
+                    '^    path: mp$']
+
+    for eline_re, aline in zip(expected_res, actual):
+        assert re.match(eline_re, aline) is not None, (aline, eline_re)
```

Comment:
At first, this (wrongly) felt overkill. But now I think this is actually really nice and pretty generic. Can you "promote" this duplicated `for` loop to a named function? Something like `match_multiline_regex()` or something.

The only potential "indirection" issue I foresee: make sure it's still possible to quickly tell which test failed when one does. If needed then maybe let the `match_multiline_regex()` pass a (sub)test name as a string. Hopefully not even needed.

Oh, and you did check that `pytest` does not have this already, right?

Reply:
I created a helper method for this, didn't find any `pytest` out-of-the-box helper.